AdobeStockImage/Model/GetImageList.php

```diff
@@ -40,42 +41,83 @@ class GetImageList implements GetImageListInterface
      */
     private $filterBuilder;
 
+    /**
+     * @var ConfigInterface $config
+     */
+    private $config;
+
     /**
      * GetImageList constructor.
      *
      * @param FilterGroupBuilder $filterGroupBuilder
      * @param GetAssetListInterface $getAssetList
      * @param FilterBuilder $filterBuilder
+     * @param ConfigInterface
      * @param array $defaultFilters
      */
     public function __construct(
         FilterGroupBuilder $filterGroupBuilder,
         GetAssetListInterface $getAssetList,
         FilterBuilder $filterBuilder,
+        ConfigInterface $configInterface,
         array $defaultFilters = []
     ) {
         $this->getAssetList = $getAssetList;
         $this->filterGroupBuilder = $filterGroupBuilder;
         $this->defaultFilters = $defaultFilters;
         $this->filterBuilder = $filterBuilder;
+        $this->config = $configInterface;
     }
 
     /**
      * @inheritdoc
      */
     public function execute(SearchCriteriaInterface $searchCriteria): SearchResultInterface
     {
+        $searchCriteria = $this->setDefaultGallery($searchCriteria);
         $searchCriteria = $this->setDefaultFilters($searchCriteria);
+
         return $this->getAssetList->execute($searchCriteria);
     }
 
+    /**
+     * Set 'gallery_id' filter if not set other 'words' filter.
+     *
+     * @param SearchCriteriaInterface $searchCriteria
+     * @return SearchCriteriaInterface
+     */
+    private function setDefaultGallery(SearchCriteriaInterface $searchCriteria): SearchCriteriaInterface
+    {
+        $isSetWordsFilter = false;
+        $filterGroups = $searchCriteria->getFilterGroups();
+        foreach ($filterGroups as $filterGroup) {
+            foreach ($filterGroup->getFilters() as $filter) {
+                if ($filter->getField() === 'words') {
+                    $isSetWordsFilter = true;
+                }
+            }
+        }
+
+        if (!$isSetWordsFilter) {
+            $galleryFilter[] = $this->filterBuilder
+                ->setField('gallery_id')
+                ->setConditionType('eq')
+                ->setValue($this->config->getDefaultGalleryId())
```

Comment:
Please check if default gallery id is set with `!empty($this->config->getDefaultGalleryId())` before applying the filter